terraform_files/vsphere/main.tf

```diff
@@ -0,0 +1,117 @@
+terraform {
+  required_providers {
+    vsphere = {
+      source = "hashicorp/vsphere"
+      version = "2.0.2"
+    }
+  }
+}
+
+locals {
+  hasISO = var.iso_download_path != "dummy" && var.iso_download_path != null
+}
+
+provider "vsphere" {
+  user                 = var.vsphere_username
+  password             = var.vsphere_password
+  vsphere_server       = var.vsphere_vcenter
+  allow_unverified_ssl = true
+}
+
+data "vsphere_datacenter" "datacenter" {
+  name = var.vsphere_datacenter
+}
+
+data "vsphere_compute_cluster" "cluster" {
+  name          = var.vsphere_cluster
+  datacenter_id = data.vsphere_datacenter.datacenter.id
+}
+
+data "vsphere_datastore" "datastore" {
+  name          = var.vsphere_datastore
+  datacenter_id = data.vsphere_datacenter.datacenter.id
+}
+
+data "vsphere_network" "network" {
+  name          = var.vsphere_network
+  datacenter_id = data.vsphere_datacenter.datacenter.id
+}
+
+# Tagging every VirtualMachine, ResourcePool, Folder with the cluster name to easy recognition and add the description:
+# "Created by the test-infra, do not delete manually"
+resource "vsphere_tag_category" "category" {
+  name        = "assisted-test-infra-${var.cluster_name}"
```

Comment:
`cluster_name` already includes `test-infra`

Reply:
no, it's not